src/pages/index.js

```diff
@@ -29,54 +32,153 @@ const enableValidation = (config) => {
 
 enableValidation(config);
 
-const createCard = (cardItem) => {
-  const card = new Card(cardItem, "#card-template", () => {
-    console.log(cardItem);
-    imgPopup.open(cardItem);
+//API
+
+export const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  authorization: "7dcd9a93-149c-4e56-87db-9285c9177a9e",
+});
+
+api
+  .getUserData()
+  .then((results) => results.json())
+  .then((data) => {
+    data;
+    currentAvatar = data.avatar;
+    initialUserData = new UserInfo(data);
   });
+
+api.getInitialCards().then((data) => {
+  data;
+
+  cardSection = new Section(
+    {
+      data: data,
+      renderer: (data) => {
+        const cardElement = createCard(data);
+        cardSection.addItem(cardElement);
+      },
+    },
+    cardListElement
+  );
+  cardSection.renderItems();
+});
+
+const createCard = (cardItem) => {
+  const card = new Card(
+    cardItem,
+    "#card-template",
+    () => {
+      imgPopup.open(cardItem);
+    },
+    () => {
+      deleteConfirmPopup.open(card);
+    }
+  );
   return card.getView();
 };
 
+//VAR
+
+let cardSection;
+let initialUserData;
+let currentAvatar;
+let profileData;
+let newCardData;
+
+//Popups
+
 const imgPopup = new PopupWithImages("#card-picture-modal");
 imgPopup.setEventListeners();
 
-const cardSection = new Section(
-  {
-    data: initialCards,
-    renderer: (data) => {
-      const cardElement = createCard(data);
-      cardSection.addItem(cardElement);
-    },
-  },
-  cardListElement
+const profilePopup = new PopupWithForm(
+  "#profile-edit-modal",
+  handleProfileUpdate
 );
-cardSection.renderItems();
-
-const userInfo = new UserInfo(".profile__name", ".profile__job");
-
-const profilePopup = new PopupWithForm("#profile-edit-modal", (data) => {
-  profilePopup.close();
-  userInfo.setUserInfo(data);
-});
 profilePopup.setEventListeners();
 
-const newCardPopup = new PopupWithForm("#card-add-modal", (data) => {
-  console.log(data);
-  const cardFormElement = createCard(data);
-  cardSection.addItem(cardFormElement);
-  newCardPopup.close();
-});
+const newCardPopup = new PopupWithForm("#card-add-modal", handleAddCard);
 newCardPopup.setEventListeners();
 
+export const deleteConfirmPopup = new PopupConfirmation(
+  "#modal-delete",
+  handleDeleteConfirmation
+);
+deleteConfirmPopup.setEventListeners();
+
+const avatarPopup = new PopupWithForm("#modal-avatar", handleUpdateAvatar);
+avatarPopup.setEventListeners();
+
+//HANDLES
+
+function handleDeleteConfirmation(card, id) {
+  card.handleDeleteCard(id);
+}
+
+function handleProfileUpdate(userData) {
+  initialUserData.setUserInfo(userData);
+  profilePopup.setLoading(true);
+  api
+    .updateUserData(userData)
+    .then(() => profilePopup.close())
+    .catch((err) => console.error(`${err}, Failed to update Avatar`))
+    .finally(() => profilePopup.setLoading(false));
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```